shaders/commonProfileShaders.cpp

```diff
@@ -55,11 +56,17 @@ namespace GLTF
         ((std::istream*)png_get_io_ptr(pngPtr))->read((char*)data, length);
     }
 #endif
+	static std::unordered_map<std::string, bool> haveAlphas;		// Caching to remove up to 98% execution time
```

Comment:
Also you're correct w.r.t. the probable impact of `log n` on datasets most people will be using, but the cache coherency is almost certainly going to have an impact. Pointer chasing has a practical impact on performance.